core/src/main/scala/org/genivi/sota/core/CampaignResource.scala

```diff
@@ -72,6 +67,29 @@ class CampaignResource(namespaceExtractor: Directive1[Namespace],
     db.run(Campaigns.fetchMeta(id).map(_.namespace))
   }
 
+  def buildCampaignStats(groupId: Uuid, updateId: Uuid): Future[CampaignStatistics] = {
+    for {
+      rows <- updateService.fetchUpdateSpecRows(updateId)
+      groupSize <- deviceRegistry.fetchDevicesInGroup(groupId)
+      successCount = rows.count(r => r.status.equals(UpdateStatus.Finished))
+      failureCount = rows.count(r => r.status.equals(UpdateStatus.Failed))
+    } yield CampaignStatistics(groupId, updateId, groupSize.size, rows.size, successCount, failureCount)
+  }
+
+  def getCampaignStats(id: Campaign.Id): Route = {
+    val dbIO = db.run(Campaigns.fetchUpdateIds(id))
```

Comment:
Should probably use fetchGroups instead, you get nicer names such as "group" and "updateRequest" instead of _2 and _1 (or is it vice versa?)